lib/ws-kernel-picker.js

```diff
@@ -73,11 +77,17 @@ export default class WSKernelPicker {
       this.onGateway.bind(this)
     );
     this.previouslyFocusedElement = gatewayListing.previouslyFocusedElement;
-    gatewayListing.setItems(gateways);
-    gatewayListing.setError("Select a gateway"); // TODO(nikita): maybe don't misuse error
+    gatewayListing.selectListView.update({
+      items: gateways,
+      infoMessage: "Select a gateway"
+    });
   }
 
   onGateway(gatewayInfo) {
+    const sessionListing = new CustomListView(
+      "No sessions available",
+      this.onSession.bind(this)
+    );
```

Comment:
Moving this up improves the rendering issue slightly. Instead of needing to switch tabs to refresh the screen the focus goes back to the editor and then moving the cursor refreshes the panel.

Not a fix really, but closer and think it might give us some ideas.